test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/IntegrationTest/IntegrationTests.cs

```diff
@@ -210,6 +213,73 @@ public void MetricExportResultIsSuccess(OtlpExportProtocol protocol, string endp
+    [InlineData(OtlpExportProtocol.HttpProtobuf, ":4318/v1/logs", ExportProcessorType.Batch)]
+    [InlineData(OtlpExportProtocol.Grpc, ":4317", ExportProcessorType.Simple)]
+    [InlineData(OtlpExportProtocol.HttpProtobuf, ":4318/v1/logs", ExportProcessorType.Simple)]
+    [InlineData(OtlpExportProtocol.Grpc, ":5317", ExportProcessorType.Simple, "https")]
+    [InlineData(OtlpExportProtocol.HttpProtobuf, ":5318/v1/logs", ExportProcessorType.Simple, "https")]
+    [Trait("CategoryName", "CollectorIntegrationTests")]
+    [SkipUnlessEnvVarFoundTheory(CollectorHostnameEnvVarName)]
+    public void LogExportResultIsSuccess(OtlpExportProtocol protocol, string endpoint, ExportProcessorType exportProcessorType, string scheme = "http")
+    {
+        using EventWaitHandle handle = new ManualResetEvent(false);
+
+        var exporterOptions = new OtlpExporterOptions
+        {
+            Endpoint = new Uri($"{scheme}://{CollectorHostname}{endpoint}"),
+            Protocol = protocol,
+        };
+
+        DelegatingExporter<LogRecord> delegatingExporter = null;
+        var exportResults = new List<ExportResult>();
+        var processorOptions = new LogRecordExportProcessorOptions();
+        processorOptions.ExportProcessorType = exportProcessorType;
+        processorOptions.BatchExportProcessorOptions = new()
+        {
+            ScheduledDelayMilliseconds = ExportIntervalMilliseconds,
+        };
+
+        using var loggerFactory = LoggerFactory.Create(builder =>
+        {
+            builder
+                .AddOpenTelemetry(options => options
+                    .AddProcessor(OtlpLogExporterHelperExtensions.BuildOtlpLogExporter(
+                        exporterOptions,
+                        processorOptions,
+                        configureExporterInstance: otlpExporter =>
+                        {
+                            delegatingExporter = new DelegatingExporter<LogRecord>
+                            {
+                                OnExportFunc = (batch) =>
+                                {
+                                    var result = otlpExporter.Export(batch);
+                                    exportResults.Add(result);
+                                    handle.Set();
+                                    return result;
+                                },
+                            };
+                            return delegatingExporter;
+                        })));
+        });
+
+        var logger = loggerFactory.CreateLogger("OtlpLogExporterTests");
+        logger.LogInformation("Hello from {name} {price}.", "tomato", 2.99);
+
+        if (exporterOptions.ExportProcessorType == ExportProcessorType.Batch)
+        {
+            Assert.True(handle.WaitOne(ExportIntervalMilliseconds * 2));
+            Assert.Single(exportResults);
+            Assert.Equal(ExportResult.Success, exportResults[0]);
+        }
+
+        if (exportProcessorType == ExportProcessorType.Simple)
```

Comment:
nit: Might be worth making this defensive against any possible future additions. Thinking like...

```csharp
       if (exporterOptions.ExportProcessorType == ExportProcessorType.Batch)
       {
             ...
       }
       else if (exportProcessorType == ExportProcessorType.Simple)
       {
             ...
       }
       else
       {
             throw new NotSupportedException("Unexpected processor type encountered.");
       }
```

A `case` might be more concise but I think you catch my drift.

Reply:
done